keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/DependsOnConstraintEvaluator.kt

```diff
@@ -19,18 +22,20 @@ import com.netflix.spinnaker.keel.core.api.DependsOnConstraint
 import com.netflix.spinnaker.keel.persistence.ArtifactRepository
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Component
+import java.time.Clock
 
 @Component
 class DependsOnConstraintEvaluator(
   private val artifactRepository: ArtifactRepository,
   private val verificationRepository: VerificationRepository,
-  override val eventPublisher: EventPublisher
-) : ConstraintEvaluator<DependsOnConstraint> {
-
+  override val eventPublisher: EventPublisher,
+  private val clock: Clock
+) : StatelessConstraintEvaluator<DependsOnConstraint, DependsOnConstraintAttributes>() {
```

Comment:
Thought for a future refactor: instead of having 2 generic types, which are intimately related, we could have a `attributeType` field in `Constraint` which returns that type. Then you'd only need the first generic type here. Same for `StatefulConstraintEvaluator` too.